### ISSUE TITLE
JOIN USING support

### ISSUE THREAD

**Issue by max-hoffman** [dolthub/dolt#5789]
A USING clause can be used in place of a join ON condition. From the [mysql docs](https://dev.mysql.com/doc/refman/8.0/en/join.html#:~:text=The%20USING(join_column_list)%20clause%20names%20a%20list%20of%20columns%20that%20must%20exist%20in%20both%20tables.%20If%20tables%20a%20and%20b%20both%20contain%20columns%20c1%2C%20c2%2C%20and%20c3%2C%20the%20following%20join%20compares%20corresponding%20columns%20from%20the%20two%20tables%3A):

> The USING(join_column_list) clause names a list of columns that must exist in both tables. If tables a and b both contain columns c1, c2, and c3, the following join compares corresponding columns from the two tables:
> ```a LEFT JOIN b USING (c1, c2, c3)```

This is conceptually similar to NATURAL JOINS, where you expect tables to share columns with the same name. Except here you specify a subset of columns. Values are backfilled with NULLs for outer joins with no column match.

**Comment by max-hoffman** [comment]
I took another look and it seems like we have parser support, but are missing the handoff in GMS. We would want `parser.go` and `sql/planbuilder/from.go` on the new path to convert `condition.Using`. The join filter will be a conjunction of equality predicates matching `leftRel.attr_i = rightRel.attr_i` for all `attr_i` in the using columns.

**Comment by jdclarke5** [comment]
As a use case, I attempted to use [pgloader](https://github.com/dimitri/pgloader/tree/master) to move a Dolt database to Postgres (reason being I like Dolt for incremental data sync, but I like Postgres for faster query execution). I found out that pgloader uses `USING` to find information about the tables. A query it performs returns an error from Dolt: `unsupported feature: USING clause on join`. See e.g. https://github.com/dimitri/pgloader/blob/master/src/sources/mysql/sql/list-all-columns.sql.
